AllReadyApp/Web-App/AllReady/Areas/Admin/Controllers/CampaignController.cs

```diff
@@ -117,11 +119,16 @@ public IActionResult Create()
                         ModelState.AddModelError("ImageUrl", "You must upload a valid image file for the logo (.jpg, .png, .gif)");
                         return View(campaign);
                     }
+
+                    if (existingImageUrl != null)
+                    {
+                        await _imageService.DeleteImageAsync(existingImageUrl);
+                    }
                 }

```

Comment:
Pretty much, although I think we need to store the result of the imageUpload call to a local var and check that for null/empty. If it has a value we can then update the campaignEvent.ImageUrl and perform the delete. With the above, if the upload failed here we'd be setting the imageUrl in the db to null so effectively still removing the image from the event.

Something like this...

```
if (fileUpload.IsAcceptableImageContentType())
{
    var existingImageUrl = campaignEvent.ImageUrl;
    var newImageUrl = await _imageService.UploadEventImageAsync(campaign.OrganizationId, campaignEvent.Id, fileUpload);
    if (!string.IsNullOrEmpty(newImageUrl))
	{
		campaignEvent.ImageUrl = newImageUrl;
		if (existingImageUrl != null)
		{
			await _imageService.DeleteImageAsync(existingImageUrl);
		}
	}
}
```